web/packages/teleterm/src/services/appUpdater/appUpdater.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  autoUpdater,
+  AppUpdater as ElectronAppUpdater,
+} from 'electron-updater';
+import { ProviderRuntimeOptions } from 'electron-updater/out/providers/Provider';
+
+import Logger from 'teleterm/logger';
+
+import { ClientToolsUpdateProvider } from './clientToolsUpdateProvider';
+
+export class AppUpdater {
+  private readonly logger = new Logger('AppUpdater');
+
+  constructor() {
+    autoUpdater.setFeedURL({
+      provider: 'custom',
+      // Wraps ClientToolsUpdateProvider to allow passing getClientToolsVersion.
+      updateProvider: class extends ClientToolsUpdateProvider {
+        constructor(
+          options: unknown,
+          updater: ElectronAppUpdater,
+          runtimeOptions: ProviderRuntimeOptions
+        ) {
+          super(() => undefined, updater, runtimeOptions);
+        }
+      },
+    });
+
+    autoUpdater.logger = this.logger;
+    autoUpdater.autoDownload = true;
+    autoUpdater.allowDowngrade = true;
+    autoUpdater.autoInstallOnAppQuit = true;
+    // Allows update checks during dev mode.
+    // Installing updates will not work.
+    // Downloads are saved to the path specified in dev-app-update.yml.
+    autoUpdater.forceDevUpdateConfig = true;
```

Comment:
I don't understand the connection between the first and the second line. If installing updates will not work, why is it being enabled in the first place?

I'm not disputing whether this is needed, it's just that I wouldn't be able to explain it based on this comment if I looked at it a year from now. ;)

Reply:
It allows us to update downloads in the dev mode, giving us at least one additional state that we can test without having to package the app. I think there's some value in it.